internal/builders/maven/action.yml

```diff
@@ -58,11 +58,21 @@ runs:
       with:
         distribution: temurin
         java-version: ${{ fromJson(inputs.slsa-workflow-inputs).jdk-version }}
+    - name: Checkout the tool repository
+      uses: slsa-framework/slsa-github-generator/.github/actions/secure-builder-checkout@main
+      with:
+        repository: slsa-framework/slsa-github-generator
+        ref: main
+        path: __TOOL_CHECKOUT_DIR__
     - name: Run mvn package
       shell: bash
       env:
         SLSA_OUTPUTS_ARTIFACTS_FILE: ${{ inputs.slsa-layout-file }}
-      run: cd plugin && mvn clean install && cd .. && mvn package
+      run: |
+        mv ./__TOOL_CHECKOUT_DIR__ ../__TOOL_CHECKOUT_DIR__
```

Comment:
I mainly copy pasted from here: https://github.com/slsa-framework/slsa-github-generator/blob/c683687ddccd1f90af85c58fa16a6ab260ea3fef/.github/workflows/delegator_lowperms-generic_slsa3.yml#L152-L158

I think at this time we are in the root of the project directory, and there might be some interference from adding a sub directory.

Reply:
if you checkout the builder first, and then checkout the project it should work with no interference, since the builder would live at GITHUB_WORKSPACE/../.

nit: use `__BUILDER_CHECKOUT_DIR__` instead of `__TOOL_CHECKOUT_DIR__`

Also don't use `main` for ref. Use v1.8.0 for now.

Reply:
Removed `mv` call.